classes/publication/PKPPublication.inc.php

```diff
@@ -131,14 +131,15 @@ public function getTitles() {
 	 * Eg - Daniel Barnes, Carlo Corino (Author); Alan Mwandenga (Translator)
 	 *
 	 * @param array $userGroups List of UserGroup objects
+	 * @param bool $preferred If the preferred public name should be used, if exists
+	 * @param bool $onlyIncludeInBrowse True iff only browse list authors should be included
 	 * @return string
 	 */
-	public function getAuthorString($userGroups) {
+	public function getAuthorString($userGroups, bool $preferred = true, bool $onlyIncludeInBrowse = false) {
```

Comment:
This is the kind of argument clutter that I'm really keen to cut down on. Instead of adding arguments to methods to try to satisfy lots of "business requirements" at once, I'd prefer to see methods split up or brokwn down into small pieces that can be composed together to achieve business rules.

I think you're already asking Bozana about whether the `bool $preferred = true` is needed. In my opinion, it isn't. If we support preferred names we should _always_ use them in rendered outputs. The only time we shouldn't use them is when a downstream consumer requires the name be broken into parts (ie - given / family).

For `$onlyIncludeInBrowse`, I'd like to see this "business requirement" live closer to the code that requires it, rather than baking it into the `getAuthorString()` method itself. One way to do this would be to add an optional argument to pass in the authors themselves:

```php
public function getAuthorString(array $userGroups, ?array $authors)
{
    // ...
}
```

We end up with the same number of arguments. But in my view, this isolates `PKPPublication::getAuthorString()` from the specific business requirements it may be used to implement. If you're worried about reimplementing the `includeInBrowse` logic in many places, you can add a helper method `PKPPublication::getIncludeInBrowseAuthors()` to centralize that logic in one place.

I appreciate I'm nitpicking here, so feel free to ignore. The PR looks good and this is obviously not a major issue. But I do think this kind of argument clutter grows over time and we help our future selves by finding ways to avoid it.